### ISSUE TITLE
Parser crash if citationKey contains brackets [ ]

### ISSUE THREAD

**Issue by TcM1911** [apcshields/zotero-bibtex-parse#1]
Thanks for a great parser! I'm using it in a project of my own.

I found a bug when I was using it. If you have brackets in the citationKey the parser craches. 

Parsing this:

```
@article{soundararajan_[186re]_liposomal_2009,
    title = {[{186Re]Liposomal} doxorubicin (Doxil): in vitro stability, pharmacokinetics, imaging and biodistribution in a head and neck squamous cell carcinoma xenograft model},
...
}
```

Throws this:

```
Uncaught Token mismatch, expected ,, found [186re]_liposomal_2009,
    title = {[{186Re]Liposomal} doxorubicin (Doxil): in vitro stability, pharmacokinetics, imaging and biodistribution in a head and neck squamous cell carcinoma xenograft model},
...
} 
zotero-bibtex-parse.js:72
BibtexParser.match zotero-bibtex-parse.js:72
BibtexParser.entry_body zotero-bibtex-parse.js:257
BibtexParser.entry zotero-bibtex-parse.js:282
BibtexParser.bibtex zotero-bibtex-parse.js:296
exports.toJSON zotero-bibtex-parse.js:485
(anonymous function) add.js:4
n.event.dispatch jquery-2.1.1.min.js:3
r.handle jquery-2.1.1.min.js:3
```

It works fine to parse without the brackets in the citationKey:

```
@article{soundararajan_186re_liposomal_2009,
    title = {[{186Re]Liposomal} doxorubicin (Doxil): in vitro stability, pharmacokinetics, imaging and biodistribution in a head and neck squamous cell carcinoma xenograft model},
...
}
```

**Comment by apcshields** [comment]
Thanks for letting me know! I’m glad you’re finding it useful.

I’ve been working on some other improvements to the project I’m using the parser for, but I should finish those up in the next couple of days and work on making the parser itself more robust. At the very least, that will mean skipping entries it can’t parse instead of crashing.

Andrew

**Comment by TcM1911** [comment]
Do you think you can make it escape bad characters during parsing?

**Comment by apcshields** [comment]
I’ll certainly give it a try.

**Comment by apcshields** [comment]
Fixed in v0.0.14.
